README.md

```diff
@@ -100,52 +169,75 @@ and from then on a simple
+                                                 "auth_method": "password" ]])
+
+The above invocation will log the message alongside meta-data merged from the `logger`'s meta-data and the meta-data provided with the `logger.trace` call.
+
+The logging meta-data is of type `typealias Logging.Metadata = [String: Metadata.Value]` where `Metadata.Value` is the following `enum` allowing nested structures rather than just `String`s as values.
+
+    public enum MetadataValue {
+        case string(String)
+        indirect case dictionary(Metadata)
+        indirect case list([Metadata.Value])
+    }
+
+Users usually don't need to interact with the `Metadata.Value` type directly as it conforms to the `ExpressibleByStringLiteral`, `ExpressibleByStringInterpolation`, `ExpressibleByDictionaryLiteral`, and `ExpressibleByArrayLiteral` protocols and can therefore be constructed using the string, array, and dictionary literals.
+
+Examples:
+
+    logger.info("you can attach strings, lists, and nested dictionaries",
+                metadata: ["key_1": "and a string value",
+                           "key_2": ["and", "a", "list", "value"],
+                           "key_3": ["where": ["we": ["pretend", ["it": ["is", "all", "Objective C", "again"]]]]]])
+    logger[metadataKey: "keys-are-strings"] = ["but", "values", ["are": "more"]]
+    logger.warning("ok, we've seen enough now.")
+
+
+
 ### Custom `LogHandler`s
 
-Now, let's put the logging meta-data aside and focus on the how we can create and configure a custom logging backend. As seen before, `Logging.make` is what gives us a fresh logger but that raises the question what kind of logging backend will I actually get when calling `Logging.make`? The answer: It's configurable _per application_. The application -- likely in its main function -- sets up the logging backend it wishes the whole application to use. Libraries should never change the logging implementation as that should owned by the application. Setting up the `LogHandler` to be used is straightforward:
+Just like meta-data, custom `LogHandler`s are an advanced feature and users will typically just choose a pre-existing package that formats and persists/delivers the log messages in an appropriate way. Said that, the proposed package here is an API package and it will become much more useful if the community create a number of useful `LogHandler`s to say format the log messages with colour or ship them to Splunk/ELK.
+
+We have already seen before that `Logging.make` is what gives us a fresh logger but that raises the question what kind of logging backend will I actually get when calling `Logging.make`? The answer: It's configurable _per application_. The application -- likely in its main function -- sets up the logging backend it wishes the whole application to use. Libraries should never change the logging implementation as that should owned by the application. Setting up the `LogHandler` to be used is straightforward:
```

Comment:
btw, i am still not crazy about "make", but i guess its the swift way?

Reply:
@tomerd the only option we have is `makeLogger`

Reply:
Seems to be the swift-way™ from what I've hears so ok with that... 
Not sure if `makeLogger` is nicer hm; your call guys